.github/workflows/push.yml

```diff
@@ -17,6 +17,9 @@ on:
   pull_request:
     branches:
       - '**'
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
```

Comment:
You might also want to include the workflow name:

https://docs.github.com/en/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow

Reply:
Good point! I'm not planning to add this to other workflows for now (nightly should always finish) but might be good in case anything is added in the future.